tests/comb_to_cggi/add_one.mlir

```diff
@@ -1,38 +1,95 @@
-            %2 = comb.truth_table %ARG00, %ARG01, %FALSE -> 6 : ui8
-            %3 = comb.truth_table %ARG00, %FALSE, %FALSE -> 1 : ui8
-            %5 = comb.truth_table %ARG00, %ARG01, %ARG02 -> 120 : ui8
-            %6 = comb.truth_table %ARG00, %ARG01, %ARG02 -> 128 : ui8
-            %8 = comb.truth_table %6, %ARG03, %FALSE -> 6 : ui8
-            %10 = comb.truth_table %6, %ARG03, %ARG04 -> 120 : ui8
-            %11 = comb.truth_table %6, %ARG03, %ARG04 -> 128 : ui8
-            %13 = comb.truth_table %11, %ARG05, %FALSE -> 6 : ui8
-            %15 = comb.truth_table %11, %ARG05, %ARG06 -> 120 : ui8
-            %16 = comb.truth_table %11, %ARG05, %ARG06 -> 128 : ui8
-            %18 = comb.truth_table %16, %ARG07, %FALSE -> 6 : ui8
-            secret.yield %18, %15, %13, %10, %8, %5, %2, %3 : i1, i1, i1, i1, i1, i1, i1, i1
-        } -> (!secret.secret<i1>, !secret.secret<i1>, !secret.secret<i1>, !secret.secret<i1>, !secret.secret<i1>, !secret.secret<i1>, !secret.secret<i1>, !secret.secret<i1>)
-  // CHECK: return
-    func.return %0#0, %0#1, %0#2, %0#3, %0#4, %0#5, %0#6, %0#7 : !secret.secret<i1>, !secret.secret<i1>, !secret.secret<i1>, !secret.secret<i1>, !secret.secret<i1>, !secret.secret<i1>, !secret.secret<i1>, !secret.secret<i1>
+  func.func @add_one(%arg0: !secret.secret<i8>) -> !secret.secret<i8> {
+    %c1_i8 = arith.constant 1 : i8
+    %0 = secret.cast %arg0 : !secret.secret<i8> to !secret.secret<tensor<8xi1>>
+    %c0_i8 = arith.constant 0 : i8
+    %1 = arith.andi %c1_i8, %c1_i8 : i8
+    %2 = arith.shrsi %1, %c0_i8 : i8
+    %3 = arith.trunci %2 : i8 to i1
+    %c2_i8 = arith.constant 2 : i8
+    %4 = arith.andi %c1_i8, %c2_i8 : i8
+    %5 = arith.shrsi %4, %c1_i8 : i8
+    %6 = arith.trunci %5 : i8 to i1
+    %c4_i8 = arith.constant 4 : i8
+    %7 = arith.andi %c1_i8, %c4_i8 : i8
+    %8 = arith.shrsi %7, %c2_i8 : i8
+    %9 = arith.trunci %8 : i8 to i1
+    %c3_i8 = arith.constant 3 : i8
+    %c8_i8 = arith.constant 8 : i8
+    %10 = arith.andi %c1_i8, %c8_i8 : i8
+    %11 = arith.shrsi %10, %c3_i8 : i8
+    %12 = arith.trunci %11 : i8 to i1
+    %c16_i8 = arith.constant 16 : i8
+    %13 = arith.andi %c1_i8, %c16_i8 : i8
+    %14 = arith.shrsi %13, %c4_i8 : i8
+    %15 = arith.trunci %14 : i8 to i1
+    %c5_i8 = arith.constant 5 : i8
+    %c32_i8 = arith.constant 32 : i8
+    %16 = arith.andi %c1_i8, %c32_i8 : i8
+    %17 = arith.shrsi %16, %c5_i8 : i8
+    %18 = arith.trunci %17 : i8 to i1
+    %c6_i8 = arith.constant 6 : i8
+    %c64_i8 = arith.constant 64 : i8
+    %19 = arith.andi %c1_i8, %c64_i8 : i8
+    %20 = arith.shrsi %19, %c6_i8 : i8
+    %21 = arith.trunci %20 : i8 to i1
+    %c7_i8 = arith.constant 7 : i8
+    %c-128_i8 = arith.constant -128 : i8
+    %22 = arith.andi %c1_i8, %c-128_i8 : i8
+    %23 = arith.shrsi %22, %c7_i8 : i8
+    %24 = arith.trunci %23 : i8 to i1
+    %from_elements = tensor.from_elements %3, %6, %9, %12, %15, %18, %21, %24 : tensor<8xi1>
```

Comment:
nit: I think if you add `--canonicalize` to the pass (or the lit test run command) it will fold this op away, since the output ends up re-extracting the individual bits to trivial_encrypt them.

Reply:
True! Updated the testcase.